ratex/core/distributed/ratex_fully_sharded_data_parallel.py

```diff
@@ -0,0 +1,85 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Module wrapper implementing ZeRO-1 in an FSDP style interface"""
+import torch.nn as nn
+import torch.nn.functional as F
+
+from raf import distributed as dist
+
+import ratex.optimizer as optim
+from ratex.core.lazy_model import all_gather, all_reduce, REDUCE_SUM
+
+
+class RatexFullyShardedDataParallel(nn.Module):
+    """FSDP ZeRO-1 wrapper"""
+
+    def __init__(self, module, optimizer=optim.SGD, optimizer_config=None):
+        super().__init__()
+        comm = dist.get_communicator()
+        self.rank = comm.rank
+        self.world_size = comm.size
+        self.module = module
+        self.params = list(self.module.parameters())
+
+        # Shard parameters for use in optimizer
+        self._shard_parameters()
+        # Optimizer initialization
+        self.optimizer = optimizer(self.sharded_parameters(), **optimizer_config or {})
+
+    def _shard_parameters(self):
+        """
+        Create list of this ranks shards
+        Shards are stored in a tuple with the respective parameter ie (parameter, shard)
+        """
+        self.sharded_params = []
+        for param in self.params:
+            shard_data = self._get_shard(param.data)
+            shard = nn.Parameter(shard_data, requires_grad=param.requires_grad)
+            self.sharded_params.append(shard)
+
+    def _get_shard(self, tensor):
```

Comment:
This function is not informative. I'd suggest `_shard_tensor` or something like that.

Reply:
Changed the method name to _shard_tensor